### PR TITLE
config: use Release binaries when building with --with-debug if that's all that's available

### LINKED ISSUE #1333: release 1.12.34.1 not compatible with --with-debug
The 1.12 release tarballs no longer contain debug binaries.  This was an intentional change, because these binaries are much larger to download and (we believe).  If people need PSOL built with debugging symbols they can use the new `--psol-from-source` option in the install script to build their own PSOL.  But the current implementation conflicts with `--with-debug`:

```
  checking for psol ... not found
  ./configure: error: module ngx_pagespeed requires the pagespeed optimization library.
  Look in /home/jefftk/nginx-1.11.6/objs/autoconf.err for more details.
  Error: Failure running './configure --add-module=/home/jefftk/ngx_pagespeed-latest-beta --with-http_ssl_module --with-debug', exiting.
```
The error file has:

```
  cc: error: /home/jefftk/ngx_pagespeed-latest-beta/psol/lib/Debug/linux/x64/pagespeed_automatic.a: No such file or directory
```

(I put` --with-debug` in when the automated installer asked for additional configure flags)

### PR DESCRIPTION
**Pull request by jeffkaufman**
Fixes #1333

### CODE REVIEW

**Comment by jeffkaufman** [comment]
Verified that this lets nginx build with `--with-debug` again:

* ran an automated install with `--with-debug` until it errored out
* observed `./configure --add-module=/home/jefftk/ngx_pagespeed-latest-beta --with-http_ssl_module --with-debug` still fails when run manually
* copied this version of `config` into `/home/jefftk/ngx_pagespeed-latest-beta/`
* observed `./configure` now passes
* ran `make`, it worked

File: config [hunk 1]
```diff
@@ -144,6 +146,9 @@ if [ "$psol_binary" = "unset" ] ; then
     psol_binary="\
         $mod_pagespeed_dir/pagespeed/automatic/pagespeed_automatic.a"
   else
+    if ! [ -d "$ngx_addon_dir/psol/lib/$buildtype" ]; then
+      buildtype=Release
```
Comment (oschaaf) [inline config]: Maybe echo a warning about this auto override?

Comment (jeffkaufman) [inline config]: @hillsp is also not keen on doing this silently

Comment (jeffkaufman) [inline config]: fixed with ee0af36

**Comment by oschaaf** [comment]
lgtm

**Comment by jeffkaufman** [comment]
@hillsp any more comments?

File: config [hunk 2]
```diff
@@ -144,6 +146,22 @@ if [ "$psol_binary" = "unset" ] ; then
...
+You have set --with-debug for building nginx, but precompiled Debug binaries for
+PSOL, which ngx_pagespeed depends on, aren't available.  You can either build
+PSOL from source with debugging information, or you can just go ahead and use
+the Release binaries."
+      echo -n "
+Use the available Release binaries?"
+      read -p " [Y/n] " yn
+      if [[ "$yn" == N* || "$yn" == n* ]]; then
+        echo "Cancelled."
+        exit 0
```
Comment (hillsp) [inline config]: exit 1, surely

File: config [hunk 3]
```diff
@@ -144,6 +146,22 @@ if [ "$psol_binary" = "unset" ] ; then
     psol_binary="\
         $mod_pagespeed_dir/pagespeed/automatic/pagespeed_automatic.a"
   else
+    if ! [ -d "$ngx_addon_dir/psol/lib/$buildtype" ]; then
+      echo "
+You have set --with-debug for building nginx, but precompiled Debug binaries for
+PSOL, which ngx_pagespeed depends on, aren't available.  You can either build
```
Comment (hillsp) [inline config]: I think I'd re-phrase the second sentence to be something like "If you're trying to debug PSOL, you must build it from source. If you just want to run nginx with verbose logging, you can use the Release binaries"

Comment (jeffkaufman) [inline config]: Done

**Comment by hillsp** [comment]
LGTM

**Comment by jeffkaufman** [comment]
cherry-picked onto branch 34 as ecd323e

**Pull request merged**